pkg/auth/providers/keycloakoidc/keycloak_provider.go

```diff
@@ -0,0 +1,200 @@
+package keycloakoidc
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"strings"
+
+	gooidc "github.com/coreos/go-oidc/v3/oidc"
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/types"
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/auth/providers/common"
+	"github.com/rancher/rancher/pkg/auth/providers/oidc"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/rancher/rancher/pkg/user"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	Name      = "keycloakoidc"
+	UserType  = "user"
+	GroupType = "group"
+)
+
+type keycloakoidcProvider struct {
+	keycloakoidcClient *KClient
+	oidcProvider       oidc.Provider
+}
+
+func Configure(ctx context.Context, mgmtCtx *config.ScaledContext, userMGR user.Manager, tokenMGR *tokens.Manager) common.AuthProvider {
+	keycloakoidcClient := &KClient{
+		httpClient: &http.Client{},
```

Comment:
Do we have a way to set  request timeout? Also why don't we configure cert and private key on the client?

Reply:
After looking into this a bit more, postman had been adding the cert/key on my other calls, so I added the cert and key to all calls. 

There isn't a way to set request timeout. Do we have a specific timeout limit we want to add?